models/DataObject/ClassDefinition/Data/ManyToOneRelation.php

```diff
@@ -322,7 +322,7 @@ public function checkValidity(mixed $data, bool $omitMandatoryCheck = false, arr
             $allow = $this->allowAssetRelation($data);
         } elseif ($data instanceof DataObject\AbstractObject) {
             $allow = $this->allowObjectRelation($data);
-        } elseif (empty($data)) {
+        } elseif ($data === null) {
```

Comment:
Looks good. I think we should also check for null with mandatory check on `line 315`.

Reply:
Done